File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string): string = 
    let line = logLine.Trim()
    let stop = line.IndexOf("]: ")
    line.Substring(stop + 3).Trim()
       

let logLevel(logLine: string): string = 
    let start = logLine.IndexOf("[")
    let levelLength = logLine.IndexOf("]: ") - start - 1
    logLine.Substring(start + 1, levelLength).Trim().ToLower()

let reformat(logLine: string): string = $"{message logLine} ({logLevel logLine})"

File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        var line = logLine.Trim();
        var stop = line.IndexOf("]: ");
        return line[(stop + 3)..].Trim();
    }

    public static string LogLevel(string logLine)
    {
        var start = logLine.IndexOf("[");
        var levelLength = logLine.IndexOf("]: ") - start - 1;
        return logLine.Substring(start + 1, levelLength).Trim().ToLower();
    }

    public static string Reformat(string logLine) 
        => $"{Message(logLine)} ({LogLevel(logLine)})";
}